src/components/TodoList.tsx

```diff
@@ -0,0 +1,103 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+import { Todo } from '../types/Todo';
+import { BlancTodo } from './BlancTodo';
+import { FilterBy } from './TodoFilter';
+
+type Props = {
+  todos: Todo[];
+  filterType: FilterBy;
+  isAdding: boolean;
+  todoName: string;
+  onDelete: (id: number) => void;
+  setErrorClosing: (err: boolean) => void;
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  filterType,
+  isAdding,
+  todoName,
+  onDelete,
+  setErrorClosing,
+}) => {
+  const [deletingId, setDeletingId] = useState<number | null>(null);
+
+  const filteredTodos = todos.filter(({ completed }) => {
+    switch (filterType) {
+      case FilterBy.Active:
+        return completed === false;
+
+      case FilterBy.Completed:
+        return completed === true;
+
+      case FilterBy.All:
+      default:
+        return true;
+    }
+  });
+
+  const handleTodoDelete = (todoId: number) => {
+    onDelete(todoId);
+    setDeletingId(todoId);
+    setErrorClosing(false);
+
+    setTimeout(() => setDeletingId(null), 1000);
+  };
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {filteredTodos.map(({ id, completed, title }) => {
+        return (
+          <div
+            key={id}
+            data-cy="Todo"
+            className={classNames(
+              { todo: !completed },
+              { 'todo completed': completed === true },
```

Comment:
@max-kravchenko 
```suggestion
              {
                todo: !completed;
                'todo completed': completed === true;
              },
```
like this

Reply:
you can do this in the next tasks)

Reply:
Thanks! Now I got it)